src/bridge/mod.rs

```diff
@@ -81,12 +87,19 @@ async fn start_neovim_runtime(instance: NeovimInstance) {
             exit(0);
         }
     }
-
     let settings = SETTINGS.get::<CmdLineSettings>();
 
     let should_handle_clipboard = settings.wsl || settings.server.is_some();
-    setup_neovide_specific_state(&nvim, should_handle_clipboard).await;
+    setup_neovide_specific_state(&session.neovim, should_handle_clipboard).await;
 
+    start_ui_command_handler(session.neovim.clone());
```

Comment:
It's not quite clear here that in fact the `Arc` shallow copied.

```suggestion
    start_ui_command_handler(Arc::clone(session.neovim));
```